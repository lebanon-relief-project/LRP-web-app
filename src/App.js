import "./App.css";
import React, { useState, useRef } from "react";
import { useOnClickOutside } from "./hooks";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Navbar from "./components/Navbar";
import Footer from "./components/Footer";
import HomePage from "./pages/homePage";
import AboutPage from "./pages/aboutPage";
import HelpPage from "./pages/helpPage";
import { Burger } from "./components";

export const Routes = {
  HOME: "/",
  ABOUT: "/about",
  HELP: "/help",
};

export const Pages = () => {
  return (
    <>
      <Switch>
        <Route path={Routes.ABOUT} component={AboutPage} />
        <Route path={Routes.HELP} component={HelpPage} />
        <Route path={Routes.HOME} component={HomePage} />
      </Switch>
    </>
  );
};

const App = () => {
  const [open, setOpen] = useState(false);
  const node = useRef();
  useOnClickOutside(node, () => setOpen(false));

  return (
    <>
      <Router>
        <div ref={node}>
          <Burger open={open} setOpen={setOpen} />
        </div>

        <Navbar open={open} setOpen={setOpen} />

        <Pages />
      </Router>
      <Footer />
    </>
  );
};

export default App;
