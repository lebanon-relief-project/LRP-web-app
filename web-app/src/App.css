@import "~antd/dist/antd.css";

.App {
  text-align: center;
}

body {
  font-family: "Raleway";
}

h1,
h2,
h3,
h4,
h5 {
  font-family: "Playfair Display";
  font-weight: bold;
}
