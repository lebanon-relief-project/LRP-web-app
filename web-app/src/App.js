import "./App.css";
import React, { useEffect, useState, useRef } from "react";
import { useOnClickOutside } from "./hooks";
import { BrowserRouter as Router, Switch, Route } from "react-router-dom";
import Navbar from "./components/Navbar";
import Footer from "./components/Footer";
import HomePage from "./pages/homePage";
import AboutPage from "./pages/aboutPage";
import HelpPage from "./pages/helpPage";
import MobileNavbar from "./components/MobileNavBar";

export const Routes = {
  HOME: "/",
  ABOUT: "/about",
  HELP: "/help",
};

export const Pages = () => {
  return (
    <>
      <Switch>
        <Route path={Routes.ABOUT} component={AboutPage} />
        <Route path={Routes.HELP} component={HelpPage} />
        <Route path={Routes.HOME} component={HomePage} />
      </Switch>
    </>
  );
};

const App = () => {
  const [open, setOpen] = useState(false);
  const node = useRef();
  useOnClickOutside(node, () => setOpen(false));

  const getList = () => {
    fetch("/health")
      .then((res) => res.json())
      .then((list) => console.log(list));
  };

  useEffect(() => {
    getList();
  }, []);
  return (
    <>
      <Router>
        <div ref={node}>
          <MobileNavbar open={open} setOpen={setOpen} />
        </div>
        <Navbar open={open} setOpen={setOpen} />
        <Pages />
      </Router>
      <Footer />
    </>
  );
};

export default App;
