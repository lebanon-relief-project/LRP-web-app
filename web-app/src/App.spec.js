import React from "react";
import { render } from "@testing-library/react";
import App, { Pages } from "./App";
import { MemoryRouter } from "react-router-dom";

jest.mock("./pages/homePage", (props) => {
  return (props) => <div>Home page</div>;
});

jest.mock("./pages/aboutPage", (props) => {
  return (props) => <div>About page</div>;
});

jest.mock("./pages/helpPage", (props) => {
  return (props) => <div>Help page</div>;
});

afterEach(() => {
  jest.clearAllMocks();
});

test("render home page", () => {
  const { container } = renderApp();
  expect(container).toMatchSnapshot();
});

test("render about page", () => {
  const { container } = renderApp("/about");
  expect(container).toMatchSnapshot();
});

test("render help page", () => {
  const { container } = renderApp("/help");
  expect(container).toMatchSnapshot();
});

const renderApp = (path = "/") => {
  return render(
    <MemoryRouter initialEntries={[path]}>
      <Pages />
    </MemoryRouter>
  );
};
