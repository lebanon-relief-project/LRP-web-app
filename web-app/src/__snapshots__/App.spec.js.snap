// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render about page 1`] = `
.c3 {
  font-weight: bold;
}

.c2 {
  padding: 1rem 0;
  margin: 2rem 0;
}

.c4 {
  list-style-type: none;
  padding: 0;
}

.c5 {
  -webkit-text-decoration-line: underline;
  text-decoration-line: underline;
}

.c10 {
  color: #000000;
}

.c8 {
  margin: 5%;
}

.c9 {
  width: 100%;
  height: auto;
}

.c6 {
  padding: 1rem 0;
  margin: 2rem 0rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 60%;
  margin-left: 20%;
}

.c0 {
  background: #f0f0f0;
  margin: 0;
  min-height: 5%;
}

@media (max-width:576px) {
  .c2 {
    width: 75%;
    padding: 0;
    margin: 0;
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }
}

@media (max-width:576px) {
  .c7 {
    padding: 0;
    margin: 0;
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 75%;
    margin: 0% 12.5%;
  }
}

@media (max-width:576px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <div>
    About page
  </div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Lebanon Relief Network
        </h2>
        <ul
          class="c4"
        >
          <li
            class="c5"
          >
            <a
              href="/terms"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              Terms of Use
            </a>
          </li>
          <li
            class="c5"
          >
            <a
              href="/privacy"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              Privacy policy
            </a>
          </li>
          <li
            class="c5"
          >
            <a
              href="https://github.com/lebanon-relief-project/LRP-web-app/blob/master/CONTRIBUTING.md"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              Code Contribution
            </a>
          </li>
        </ul>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <a
            class="c8"
            href="https://www.instagram.com/lebanonreliefnetwork"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Instagram"
              class="c9"
              src="Instagram.svg"
            />
          </a>
          <a
            class="c8"
            href="https://www.facebook.com/Lebanonreliefnetwork"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Facebook"
              class="c9"
              src="Facebook.svg"
            />
          </a>
          <a
            class="c8"
            href="https://twitter.com/LebReliefNet"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Twitter"
              class="c9"
              src="Twitter.svg"
            />
          </a>
        </div>
        <a
          class="c10"
          href="mailto:contact@lebanonreliefnetwork.org"
          target="_blank"
        >
          contact@lebanonreliefnetwork.org
        </a>
      </div>
    </div>
  </section>
</div>
`;

exports[`render help page 1`] = `
.c3 {
  font-weight: bold;
}

.c2 {
  padding: 1rem 0;
  margin: 2rem 0;
}

.c4 {
  list-style-type: none;
  padding: 0;
}

.c5 {
  -webkit-text-decoration-line: underline;
  text-decoration-line: underline;
}

.c10 {
  color: #000000;
}

.c8 {
  margin: 5%;
}

.c9 {
  width: 100%;
  height: auto;
}

.c6 {
  padding: 1rem 0;
  margin: 2rem 0rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 60%;
  margin-left: 20%;
}

.c0 {
  background: #f0f0f0;
  margin: 0;
  min-height: 5%;
}

@media (max-width:576px) {
  .c2 {
    width: 75%;
    padding: 0;
    margin: 0;
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }
}

@media (max-width:576px) {
  .c7 {
    padding: 0;
    margin: 0;
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 75%;
    margin: 0% 12.5%;
  }
}

@media (max-width:576px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <div>
    Help page
  </div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Lebanon Relief Network
        </h2>
        <ul
          class="c4"
        >
          <li
            class="c5"
          >
            <a
              href="/terms"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              Terms of Use
            </a>
          </li>
          <li
            class="c5"
          >
            <a
              href="/privacy"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              Privacy policy
            </a>
          </li>
          <li
            class="c5"
          >
            <a
              href="https://github.com/lebanon-relief-project/LRP-web-app/blob/master/CONTRIBUTING.md"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              Code Contribution
            </a>
          </li>
        </ul>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <a
            class="c8"
            href="https://www.instagram.com/lebanonreliefnetwork"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Instagram"
              class="c9"
              src="Instagram.svg"
            />
          </a>
          <a
            class="c8"
            href="https://www.facebook.com/Lebanonreliefnetwork"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Facebook"
              class="c9"
              src="Facebook.svg"
            />
          </a>
          <a
            class="c8"
            href="https://twitter.com/LebReliefNet"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Twitter"
              class="c9"
              src="Twitter.svg"
            />
          </a>
        </div>
        <a
          class="c10"
          href="mailto:contact@lebanonreliefnetwork.org"
          target="_blank"
        >
          contact@lebanonreliefnetwork.org
        </a>
      </div>
    </div>
  </section>
</div>
`;

exports[`render home page 1`] = `
.c3 {
  font-weight: bold;
}

.c2 {
  padding: 1rem 0;
  margin: 2rem 0;
}

.c4 {
  list-style-type: none;
  padding: 0;
}

.c5 {
  -webkit-text-decoration-line: underline;
  text-decoration-line: underline;
}

.c10 {
  color: #000000;
}

.c8 {
  margin: 5%;
}

.c9 {
  width: 100%;
  height: auto;
}

.c6 {
  padding: 1rem 0;
  margin: 2rem 0rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 60%;
  margin-left: 20%;
}

.c0 {
  background: #f0f0f0;
  margin: 0;
  min-height: 5%;
}

@media (max-width:576px) {
  .c2 {
    width: 75%;
    padding: 0;
    margin: 0;
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }
}

@media (max-width:576px) {
  .c7 {
    padding: 0;
    margin: 0;
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 75%;
    margin: 0% 12.5%;
  }
}

@media (max-width:576px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <div>
    Home page
  </div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Lebanon Relief Network
        </h2>
        <ul
          class="c4"
        >
          <li
            class="c5"
          >
            <a
              href="/terms"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              Terms of Use
            </a>
          </li>
          <li
            class="c5"
          >
            <a
              href="/privacy"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              Privacy policy
            </a>
          </li>
          <li
            class="c5"
          >
            <a
              href="https://github.com/lebanon-relief-project/LRP-web-app/blob/master/CONTRIBUTING.md"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              Code Contribution
            </a>
          </li>
        </ul>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <a
            class="c8"
            href="https://www.instagram.com/lebanonreliefnetwork"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Instagram"
              class="c9"
              src="Instagram.svg"
            />
          </a>
          <a
            class="c8"
            href="https://www.facebook.com/Lebanonreliefnetwork"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Facebook"
              class="c9"
              src="Facebook.svg"
            />
          </a>
          <a
            class="c8"
            href="https://twitter.com/LebReliefNet"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Twitter"
              class="c9"
              src="Twitter.svg"
            />
          </a>
        </div>
        <a
          class="c10"
          href="mailto:contact@lebanonreliefnetwork.org"
          target="_blank"
        >
          contact@lebanonreliefnetwork.org
        </a>
      </div>
    </div>
  </section>
</div>
`;

exports[`render results page 1`] = `
.c3 {
  font-weight: bold;
}

.c2 {
  padding: 1rem 0;
  margin: 2rem 0;
}

.c4 {
  list-style-type: none;
  padding: 0;
}

.c5 {
  -webkit-text-decoration-line: underline;
  text-decoration-line: underline;
}

.c10 {
  color: #000000;
}

.c8 {
  margin: 5%;
}

.c9 {
  width: 100%;
  height: auto;
}

.c6 {
  padding: 1rem 0;
  margin: 2rem 0rem;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 60%;
  margin-left: 20%;
}

.c0 {
  background: #f0f0f0;
  margin: 0;
  min-height: 5%;
}

@media (max-width:576px) {
  .c2 {
    width: 75%;
    padding: 0;
    margin: 0;
    -webkit-order: 2;
    -ms-flex-order: 2;
    order: 2;
  }
}

@media (max-width:576px) {
  .c7 {
    padding: 0;
    margin: 0;
    -webkit-order: 1;
    -ms-flex-order: 1;
    order: 1;
  }
}

@media (max-width:768px) {
  .c1 {
    width: 75%;
    margin: 0% 12.5%;
  }
}

@media (max-width:576px) {
  .c1 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
    -webkit-align-items: center;
    -webkit-box-align: center;
    -ms-flex-align: center;
    align-items: center;
  }
}

<div>
  <div>
    Results page
  </div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Lebanon Relief Network
        </h2>
        <ul
          class="c4"
        >
          <li
            class="c5"
          >
            <a
              href="/terms"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              Terms of Use
            </a>
          </li>
          <li
            class="c5"
          >
            <a
              href="/privacy"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              Privacy policy
            </a>
          </li>
          <li
            class="c5"
          >
            <a
              href="https://github.com/lebanon-relief-project/LRP-web-app/blob/master/CONTRIBUTING.md"
              rel="noopener noreferrer"
              style="text-decoration: none;"
              target="_blank"
            >
              Code Contribution
            </a>
          </li>
        </ul>
      </div>
      <div
        class="c6"
      >
        <div
          class="c7"
        >
          <a
            class="c8"
            href="https://www.instagram.com/lebanonreliefnetwork"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Instagram"
              class="c9"
              src="Instagram.svg"
            />
          </a>
          <a
            class="c8"
            href="https://www.facebook.com/Lebanonreliefnetwork"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Facebook"
              class="c9"
              src="Facebook.svg"
            />
          </a>
          <a
            class="c8"
            href="https://twitter.com/LebReliefNet"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Twitter"
              class="c9"
              src="Twitter.svg"
            />
          </a>
        </div>
        <a
          class="c10"
          href="mailto:contact@lebanonreliefnetwork.org"
          target="_blank"
        >
          contact@lebanonreliefnetwork.org
        </a>
      </div>
    </div>
  </section>
</div>
`;
